src/lib.rs

```
@@ -269,8 +269,16 @@ impl ObjectStore for HdfsObjectStore {
             .read(&make_absolute_file(location))
             .await
             .to_object_store_err()?;
+        let start: usize = range
+            .start
+            .try_into()
+            .expect("unable to convert range.start to usize");
```

What's the difference between this and just
```rust
let start = range.start as usize
```
that's probably all I would have done. I assume that's just like an implicit `try_into().unwrap()`

---

I guess on the object meta the `try_into` will automatically pick the right type for the different versions, but since multiple versions aren't actually tested it might make sense to just update the `object_store` version to ~0.12. This is a slightly larger change then the 0.10 -> 0.11 bump

---

sure, I think this is up to you! probably safest is just releasing a new version of this crate with only object_store 0.12?

---

> What's the difference between this and just
> 
> ```rust
> let start = range.start as usize
> ```
> 
> that's probably all I would have done. I assume that's just like an implicit `try_into().unwrap()`

the `as` cast will silently truncate on platforms with <64 bit usize